### PR TITLE
package for frama-c.24.0

### PR DESCRIPTION
**Pull request by vprevosto**
Update Frama-C package to latest release

### CODE REVIEW

**Comment by mseri** [comment]
Is the failure of the naked pointers checker on the tests expected?

the other failure is due to dune 1.11 in the lower bounds, unrelated to this PR:
```

#=== ERROR while compiling dune.1.11.0 ========================================#
# context              2.1.1 | linux/x86_64 | ocaml-base-compiler.4.10.2 | file:///home/opam/opam-repository
# path                 ~/.opam/4.10/.opam-switch/build/dune.1.11.0
# command              ~/.opam/opam-init/hooks/sandbox.sh build ./boot.exe --release -j 31
# exit-code            1
# env-file             ~/.opam/log/dune-99-bf8fcb.env
# output-file          ~/.opam/log/dune-99-bf8fcb.out
### output ###
# Error: stat: /home/opam/.opam/4.10/lib/ocaml/arith_flags.cmx: No such file or directory
# -> required by src/stdune/.stdune.objs/native/stdune__Appendable_list.cmx
# -> required by src/catapult/.catapult.objs/native/catapult.cmx
# -> required by src/.dune.objs/native/dune__Action.cmx
# -> required by bin/.main.objs/native/main.cmx
# -> required by bin/main.a
# -> required by bin/main/main_dune.exe
# -> required by install bin/dune
# -> required by dune.install
```

sorry for being so slow in the other PR

**Comment by vprevosto** [comment]
> Is the failure of the naked pointers checker on the tests expected?

kind of. Apparently the fact that Frama-C has naked pointers is a well-known fact in the OCaml community, since it is used as example in the [post](https://discuss.ocaml.org/t/ann-a-dynamic-checker-for-detecting-naked-pointers/5805) introducing nnp-checker 😬, and I have to admit that we haven't really tried to mitigate the situation so far. 

I imagine we should add a conflict on these compiler variants, then?

> the other failure is due to dune 1.11 in the lower bounds, unrelated to this PR:
> 

yes, I have seen that on the other PR, and I have to admit that I wasn't sure there was something that could be done at Frama-C level.

> 
> sorry for being so slow in the other PR

No problem. On the contrary, we have to thank the whole opam team for taking care of all these packages.

File: packages/frama-c/frama-c.24.0/opam [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  # Coq: because .vo would would not be loadable by another version of Coq
+  # libraries: because we use dynamic linking
+  "apron"
+  "coq"
+  "mlgmpidl"
+  "ppx_deriving"
+  "ppx_deriving_yojson"
+  "zmq"
+]
+
```
Comment (mseri) [inline packages/frama-c/frama-c.24.0/opam:139]:
```suggestion

x-ci-accept-failures: [
  "nnp" "nnpchecker"
  "centos-7"  # Too old version of make
]

```

File: packages/frama-c/frama-c.24.0/opam [hunk 2]
```diff
@@ -0,0 +1,159 @@
+opam-version: "2.0"
+name: "frama-c"
```
Comment (mseri) [inline packages/frama-c/frama-c.24.0/opam:2]:
```suggestion
```

File: packages/frama-c/frama-c.24.0/opam [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+synopsis: "Platform dedicated to the analysis of source code written in C"
+version: "24.0"
```
Comment (mseri) [inline packages/frama-c/frama-c.24.0/opam:4]:
```suggestion
```

**Pull request merged**